agent/structs/config_entry_mesh.go

```diff
@@ -25,6 +28,20 @@ type TransparentProxyMeshConfig struct {
 	MeshDestinationsOnly bool `alias:"mesh_destinations_only"`
 }
 
+type MeshTLSConfig struct {
+	Incoming *MeshDirectionTLSConfig `json:",omitempty"`
+	Outgoing *MeshDirectionTLSConfig `json:",omitempty"`
+}
+
+type MeshDirectionTLSConfig struct {
```

Comment:
```suggestion
type MeshDirectionalTLSConfig struct {
```
Nitpick, but I think this would make it slightly more clear that it's an adjective describing which TLS config rather than the subject being configured.